### PR TITLE
*: add a session variable for table partition, tiny update

### PR DESCRIPTION
**Pull request by tiancaiamao**
## What have you changed? (mandatory)

- add a 'tidb_enable_table_partition' session variable
- update a TODO

## What are the type of the changes (mandatory)?

- Improvement (non-breaking change which is an improvement to an existing feature)

## How has this PR been tested (mandatory)?

unit test

## Add a few positive/negative examples (optional)

@winoros @coocood @XuHuaiyu

### CODE REVIEW

File: sessionctx/variable/sysvar.go [hunk 1]
```diff
@@ -636,6 +636,7 @@ var defaultSysVars = []*SysVar{
 	{ScopeSession, TIDBMemQuotaNestedLoopApply, strconv.FormatInt(DefTiDBMemQuotaNestedLoopApply, 10)},
 	{ScopeSession, TiDBEnableStreaming, "0"},
 	{ScopeSession, TxnIsolationOneShot, ""},
+	{ScopeSession, TiDBEnableTablePartition, "0"},
```
Comment (jackysp) [inline sessionctx/variable/sysvar.go]: Is it safe to use a session variable to control it? I think it should be a global variable and require to restart the server.

File: table/tables/tables_test.go [hunk 2]
```diff
@@ -338,7 +338,9 @@ PARTITION BY RANGE ( id ) (
 		PARTITION p3 VALUES LESS THAN (21)
 )`
 
-	_, err := ts.se.Execute(context.Background(), createTable1)
+	_, err := ts.se.Execute(context.Background(), "set @@session.tidb_enable_table_partition=1")
```
Comment (coocood) [inline table/tables/tables_test.go]: So we don't need to set `tbInfo.Partition.Enable = true`?

File: sessionctx/variable/sysvar.go (hunk 1 above)
Comment (tiancaiamao) [reply sessionctx/variable/sysvar.go]: It's added for testing... we can't enable it before table partition is fully implemented.

File: table/tables/tables_test.go (hunk 2 above)
Comment (tiancaiamao) [reply table/tables/tables_test.go]: Done.

File: ddl/ddl_api.go [hunk 3]
```diff
@@ -842,6 +842,9 @@ func (d *ddl) CreateTable(ctx sessionctx.Context, s *ast.CreateTableStmt) (err e
 		pi := &model.PartitionInfo{
 			Type: s.Partition.Tp,
 		}
+		if ctx.GetSessionVars().EnableTablePartition {
+			pi.Enable = true
```
Comment (zhexuany) [inline ddl/ddl_api.go]: how about `pi.Enable = ctx.GetSessionVars().EnableTablePartition` ?

Comment (tiancaiamao) [reply ddl/ddl_api.go]: Changed. PTAL @zhexuany

**Comment by tiancaiamao** [comment]
PTAL @coocood @jackysp @zhexuany

**Pull request merged**